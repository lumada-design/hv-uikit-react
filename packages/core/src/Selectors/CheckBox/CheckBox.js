/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useState } from "react";
import PropTypes from "prop-types";
import uniqueId from "lodash/uniqueId";
import CheckBoxIcon from "@hv/uikit-react-icons/dist/Generic/Checkbox";
import CheckBoxCheckedIcon from "@hv/uikit-react-icons/dist/Generic/CheckboxCheck";
import CheckBoxPartialIcon from "@hv/uikit-react-icons/dist/Generic/CheckboxPartial";
import deprecatedPropType from "@material-ui/core/utils/deprecatedPropType";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import classNames from "classnames";
import labelPositions from "../labelPositions";

/**
 * Chooses the correct label styling to applied based on position.
 *
 * @param {String} classes - The classes object containing the classes names needed to be applied.
 * @param {Object} labelPosition - an Object containing the avaible label positions.
 * @returns {Object} - an Object with the name of the class for the required styling.
 */
const prepareLabelStyles = (classes, labelPosition, label) => {
  if (label) {
    switch (labelPosition) {
      default:
      case labelPositions.end:
        return classNames(classes.container, classes.labelEnd);
      case labelPositions.start:
        return classNames(classes.container, classes.labelStart);
    }
  }
  return classes.container;
};

/**
 * Chooses the correct icon to used based on the disable value.
 *
 * @param {Object} props - HvCheckbox props.
 * @returns {Object} - an Object with the selected icons.
 */
const prepareIcon = ({ classes, theme, disabled }) => {
  const color = disabled
    ? [theme.hv.palette.atmosphere.atmo4, theme.hv.palette.atmosphere.atmo6]
    : null;

  return {
    emptyIcon: <CheckBoxIcon color={color} className={classes.icon} />,
    checkedIcon: <CheckBoxCheckedIcon color={color} className={classes.icon} />,
    indeterminateIcon: (
      <CheckBoxPartialIcon color={color} className={classes.icon} />
    )
  };
};

const HvCheckbox = props => {
  const {
    classes,
    className,
    id,
    checked,
    indeterminate,
    disabled,
    onChange,
    value,
    label,
    labelPlacement,
    formControlLabelProps,
    propsIcon,
    checkboxProps,
    propsLabel,
    ...other
  } = props;
  const materialPrimaryColor = "primary";
  const icons = prepareIcon(props);
  const labelClass = prepareLabelStyles(classes, labelPlacement, label);
  const [isFocusDisabled, disableFocus] = useState(false);
  const [internalId] = useState(id || uniqueId("hv-checkbox-"));

  const onLocalChange = evt => {
    const isKeyEvent =
      window.event.screenX === 0 &&
      window.event.screenY === 0 &&
      window.event.clientX === 0 &&
      window.event.clientY === 0;

    disableFocus(!isKeyEvent);
    onChange(evt);
  };

  const onBlur = () => {
    disableFocus(false);
  };

  return (
    <FormControlLabel
      label={label}
      labelPlacement={labelPlacement}
      disabled={disabled}
      className={classNames(labelClass, classes.truncate, className, {
        [classes.disableFocus]: isFocusDisabled
      })}
      id={internalId}
      classes={{
        disabled: classes.labelDisabled,
        label: classes.labelTypography
      }}
      control={
        <Checkbox
          id={`${internalId}-input`}
          className={classes.checkBox}
          icon={icons.emptyIcon}
          indeterminateIcon={icons.indeterminateIcon}
          checkedIcon={icons.checkedIcon}
          color={materialPrimaryColor}
          disabled={disabled}
          disableRipple
          onChange={onLocalChange}
          onBlur={onBlur}
          value={value}
          checked={checked}
          indeterminate={indeterminate}
          {...checkboxProps}
          {...propsLabel}
          {...other}
        />
      }
      {...formControlLabelProps}
      {...propsIcon}
    />
  );
};

HvCheckbox.propTypes = {
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied to the checkbox.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component.
     */
    container: PropTypes.string,
    /**
     * Styles applied to the label typography.
     */
    labelTypography: PropTypes.string,
    /**
     * Styles applied to the component when the label is disable.
     */
    labelDisabled: PropTypes.string,
    /**
     *  Styles applied to the label when the position is end.
     */
    labelEnd: PropTypes.string,
    /**
     * Styles applied to the label when the position is start.
     */
    labelStart: PropTypes.string,
    /**
     * Styles applied to the checkbox core element (material-ui).
     */
    checkBox: PropTypes.string,
    /**
     * Styles applied to the icon.
     */
    icon: PropTypes.string
  }).isRequired,
  /**
   * If `true` the checkbox is disabled and the onClick function will not be called.
   */
  disabled: PropTypes.bool,
  /**
   * The function executed when the checkbox is pressed.
   */
  onChange: PropTypes.func,
  /**
   * If `true` the checkbox is selected, if set to `false` the checkbox is not selected.
   * note: if this value is specified the state of the checkbox must be managed
   */
  checked: PropTypes.bool,
  /**
   * If `true` the checkbox uses the intermediate state, if set to `false` the checkbox will not use the intermediate state.
   */
  indeterminate: PropTypes.bool,
  /**
   * The value of the checkbox. This value will be returned in the event object generated for the onChange callback
   */
  value: PropTypes.string,
  /**
   * The label to be added to the checkbox.
   */
  label: PropTypes.string,
  /**
   * The position of the checkbox label.
   *  - Accepted values:
   *    --"start",
   *    --"end"
   *  - note: the labelPositions object should be used to set this value.
   */
  labelPlacement: PropTypes.oneOf(["start", "end"]),
  /**
   * Extra properties passed to the MUI FormControlLabel component.
   */
  formControlLabelProps: PropTypes.instanceOf(Object),
  /**
   * @deprecated Instead use the formControlLabelProps property
   */
  propsIcon: deprecatedPropType(
    PropTypes.string,
    "Instead use the formControlLabelProps property"
  ),
  /**
   * Extra properties passed to the MUI Checkbox component.
   */
  checkboxProps: PropTypes.instanceOf(Object),
  /**
   * @deprecated Instead use the checkboxProps property
   */
  propsLabel: deprecatedPropType(
    PropTypes.string,
    "Instead use the checkboxProps property"
  ),
  /**
   * The theme passed by the provider.
   */
  theme: PropTypes.instanceOf(Object)
};

HvCheckbox.defaultProps = {
  className: "",
  id: undefined,
  value: "",
  label: "",
  checked: undefined,
  indeterminate: undefined,
  disabled: false,
  onChange: () => {},
  formControlLabelProps: undefined,
  propsIcon: undefined,
  checkboxProps: undefined,
  propsLabel: undefined,
  labelPlacement: "end",
  theme: null
};

export default HvCheckbox;
