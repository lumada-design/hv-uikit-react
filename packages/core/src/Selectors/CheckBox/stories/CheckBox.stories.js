import React, { useState } from "react";
import { HvCheckBox } from "../../..";

export default {
  title: "Components/Selectors/Checkbox",
  parameters: {
    componentSubtitle: null,
    usage: 'import { HvCheckBox } from "@hitachivantara/uikit-react-core";',
  },
  component: HvCheckBox,
};

export const Main = () => {
  return <HvCheckBox id="test" inputProps={{ "aria-label": "Checkbox example" }} />;
};

export const Disabled = () => (
  <HvCheckBox id="Checkbox-disabled" disabled inputProps={{ "aria-label": "Checkbox disabled" }} />
);

Disabled.story = {
  parameters: {
    docs: {
      storyDescription: "A sample showcasing a disable checkbox that does not allows interaction.",
    },
    pa11y: {
      ignore: [
        "region",
        // Text or images of text that are part of an inactive user interface component have no contrast requirement.
        // https://github.com/lumada-design/hv-uikit-react/issues/775#issuecomment-557167364
        "WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail",
        "color-contrast",
      ],
    },
  },
};

export const WithLabel = () => <HvCheckBox id="check-label" label="Label" />;

WithLabel.story = {
  parameters: {
    docs: {
      storyDescription: "A sample showcasing a checkbox that includes a label.",
    },
  },
};

export const WithClickAction = () => (
  <HvCheckBox
    id="check-click"
    label="Click me!"
    value="A"
    onChange={(event) => console.log(`my value is ${event.target.value}`)}
  />
);

WithClickAction.story = {
  parameters: {
    docs: {
      storyDescription:
        "Returns the value when clicked, check the browser console to see the result.",
    },
  },
};

export const WithClickActionDisabled = () => (
  <HvCheckBox
    id="check-click"
    label="Click me!"
    value="A"
    disabled
    onChange={(event) => console.log(`my value is ${event.target.value}`)}
  />
);

WithClickActionDisabled.story = {
  parameters: {
    docs: {
      storyDescription:
        "Dos not return the value when clicked because is disabled, check the browser console to see the result.",
    },
    pa11y: {
      ignore: [
        "region",
        // Text or images of text that are part of an inactive user interface component have no contrast requirement.
        // https://github.com/lumada-design/hv-uikit-react/issues/775#issuecomment-557167364
        "WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail",
        "color-contrast",
      ],
    },
  },
};

export const WithState = () => {
  const [aChecked, setAChecked] = useState(false);
  const [bChecked, setBChecked] = useState(true);
  const [cChecked, setCChecked] = useState(false);

  return (
    <div
      style={{
        display: "flex",
        flexDirection: "column",
        alignItems: "flex-start",
      }}
    >
      <HvCheckBox
        id="checkState1"
        label="Checkbox 1"
        checked={aChecked}
        onChange={() => setAChecked(!aChecked)}
      />
      <HvCheckBox
        id="checkState2"
        label="Checkbox 2"
        checked={bChecked}
        onChange={() => setBChecked(!bChecked)}
      />
      <HvCheckBox
        id="checkState3"
        label="Checkbox indeterminate"
        indeterminate={cChecked}
        onChange={() => setCChecked(!cChecked)}
      />

      <HvCheckBox id="checkState4" label="Checkbox indeterminate disabled" disabled indeterminate />
      <HvCheckBox id="checkState5" label="Checkbox disabled" disabled />
      <HvCheckBox id="checkState6" label="Checkbox checked disabled" checked disabled />
    </div>
  );
};

WithState.story = {
  parameters: {
    docs: {
      storyDescription: "Demostration of how to set the state of the checkbox",
    },
    pa11y: {
      ignore: [
        "region",
        // Text or images of text that are part of an inactive user interface component have no contrast requirement.
        // https://github.com/lumada-design/hv-uikit-react/issues/775#issuecomment-557167364
        "WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail",
        "color-contrast",
      ],
    },
  },
};
