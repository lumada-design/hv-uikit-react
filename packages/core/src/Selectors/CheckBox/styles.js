/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const styles = theme => ({
  container: {
    height: "32px",
    marginRight: 0,
    marginLeft: 0,
    "&:hover": {
      backgroundColor: theme.hv.palette.atmosphere.atmo4
    }
  },
  labelTypography: {
    fontWeight: theme.typography.body1.fontWeight,
    letterSpacing: theme.typography.body1.letterSpacing,
    color: theme.typography.body1.color,
    fontSize: theme.typography.body1.fontSize,
    lineHeight: theme.typography.body1.lineHeight
  },
  labelDisabled: {
    backgroundColor: "transparent",
    cursor: "not-allowed",
    "&:hover": {
      backgroundColor: "transparent",
      cursor: "not-allowed"
    }
  },
  labelEnd: {
    paddingRight: "8px"
  },
  labelStart: {
    paddingLeft: "8px"
  },
  checkBox: {
    padding: "8px"
  },
  icon: {
    width: "16px",
    height: "16px",
    backgroundSize: "16px 16px"
  }
});

export default styles;
