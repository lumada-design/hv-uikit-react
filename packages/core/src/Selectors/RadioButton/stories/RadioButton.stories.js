import React, { useState } from "react";
import { HvRadio } from "../../..";

export default {
  title: "Components/Selectors/Radio",
  parameters: {
    componentSubtitle: null,
    usage: 'import { HvRadio } from "@hitachivantara/uikit-react-core";',
  },
  component: HvRadio,
};

export const Main = () => {
  return <HvRadio inputProps={{ "aria-label": "Simple radio button" }} />;
};

export const Disabled = () => (
  <HvRadio id="radio1" disabled inputProps={{ "aria-label": "Radio disabled" }} />
);

Disabled.story = {
  parameters: {
    docs: {
      storyDescription:
        "A sample showcasing a disabled radio button that does not allows interaction.",
    },
    pa11y: {
      ignore: [
        "region",
        // Text or images of text that are part of an inactive user interface component have no contrast requirement.
        // https://github.com/lumada-design/hv-uikit-react/issues/775#issuecomment-557167364
        "WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail",
        "color-contrast",
      ],
    },
  },
};

export const CheckedDisabled = () => (
  <HvRadio id="radio1" disabled checked inputProps={{ "aria-label": "Radio disabled" }} />
);

CheckedDisabled.story = {
  parameters: {
    docs: {
      storyDescription:
        "A sample showcasing a checked and disabled radio button that does not allows interaction.",
    },
    pa11y: {
      ignore: [
        "region",
        // Text or images of text that are part of an inactive user interface component have no contrast requirement.
        // https://github.com/lumada-design/hv-uikit-react/issues/775#issuecomment-557167364
        "WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail",
        "color-contrast",
      ],
    },
  },
};

export const Label = () => <HvRadio id="radio1" label="Label" />;

Label.story = {
  parameters: {
    docs: {
      storyDescription: "A sample showcasing a radio button that includes a label.",
    },
  },
};

export const WithClickAction = () => (
  <HvRadio
    id="check-click"
    label="Click me!"
    value="A"
    onChange={(event) => console.log(`my value is ${event.target.value}`)}
  />
);

WithClickAction.story = {
  parameters: {
    docs: {
      storyDescription:
        "Returns the value when clicked, check the browser console to see the result.",
    },
  },
};

export const WithClickActionDisabled = () => (
  <HvRadio
    id="radio1"
    label="Click me!"
    value="A"
    disabled
    onChange={(event) => console.log(`my value is ${event.target.value}`)}
  />
);

WithClickActionDisabled.story = {
  parameters: {
    docs: {
      storyDescription:
        "Dos not return the value when clicked because is disabled, check the browser console to see the result.",
    },
    pa11y: {
      ignore: [
        "region",
        // Text or images of text that are part of an inactive user interface component have no contrast requirement.
        // https://github.com/lumada-design/hv-uikit-react/issues/775#issuecomment-557167364
        "WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail",
        "color-contrast",
      ],
    },
    eyes: {
      // this sample showcases functionality, the disabled state is already showcased above
      include: false,
    },
  },
};

export const WithState = () => {
  const valueA = "a";
  const valueB = "b";
  const nothing = "";
  const [checkedRadio, setCheckedRadio] = useState(valueA);

  const onChangeRadio = (value) => () => {
    setCheckedRadio(checkedRadio !== value ? value : nothing);
  };

  return (
    <div style={{ display: "inline-flex", flexDirection: "column" }}>
      <HvRadio
        id="radio1"
        label="Radio 1"
        checked={checkedRadio === valueA}
        onChange={onChangeRadio(valueA)}
      />
      <HvRadio
        id="radio2"
        label="Radio 2"
        checked={checkedRadio === valueB}
        onChange={onChangeRadio(valueB)}
      />
    </div>
  );
};

WithState.story = {
  parameters: {
    docs: {
      storyDescription: "Demostration of how to set the state of the radio button",
    },
  },
};
