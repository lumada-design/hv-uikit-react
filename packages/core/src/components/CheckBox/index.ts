export { default as checkBoxClasses } from "./checkBoxClasses";
export * from "./checkBoxClasses";
export * from "./CheckBox";
