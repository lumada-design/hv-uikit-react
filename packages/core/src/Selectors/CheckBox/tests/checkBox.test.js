/* eslint-env jest */

// import { mount } from "enzyme";
import React from "react";
import { mount } from "enzyme";

import { Checkbox } from "@hitachivantara/uikit-react-icons";

import { HvCheckBox, HvProvider } from "../../..";
import labelPositions from "../../labelPositions";
import theme from "../../../theme";

import { Main, Disabled } from "../stories/CheckBox.stories";

describe("CheckBox", () => {
  let wrapper;

  const labelStartClassName = "labelStart";
  const labelEndClassName = "labelEnd";

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <Main />
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvCheckBox)).toMatchSnapshot();
  });

  it("should render the Checkbox component", () => {
    const buttonComponent = wrapper.find(HvCheckBox);
    expect(buttonComponent.length).toBe(1);
  });

  it("should have an icon", () => {
    const component = wrapper.find(HvCheckBox);
    const componentHtml = component.html();

    expect(component.find(Checkbox).length).toBe(1);

    expect(componentHtml.includes(theme.hv.palette.atmosphere.atmo4)).toBe(false);
    expect(componentHtml.includes(theme.hv.palette.atmosphere.atmo6)).toBe(false);
  });

  it("should have an icon when disabled", () => {
    wrapper = mount(
      <HvProvider>
        <Disabled />
      </HvProvider>
    );

    const component = wrapper.find(HvCheckBox);
    const componentHtml = component.html();

    expect(component.find(Checkbox).length).toBe(1);

    expect(componentHtml.includes(theme.hv.palette.atmosphere.atmo4)).toBe(true);
    expect(componentHtml.includes(theme.hv.palette.atmosphere.atmo6)).toBe(true);
  });

  it("should apply the correct class name when there is a label at the start", () => {
    const mountWrapper = mount(
      <HvProvider>
        <HvCheckBox label="test" labelPlacement={labelPositions.start} />
      </HvProvider>
    ).find(HvCheckBox);
    expect(mountWrapper.html().includes(labelStartClassName)).toBe(true);
    expect(mountWrapper.html().includes(labelEndClassName)).toBe(false);
  });

  it("should apply the correct class name when there is a label at the end", () => {
    const mountWrapper = mount(
      <HvProvider>
        <HvCheckBox label="test" labelPlacement={labelPositions.end} />
      </HvProvider>
    ).find(HvCheckBox);
    expect(mountWrapper.html().includes(labelStartClassName)).toBe(false);
    expect(mountWrapper.html().includes(labelEndClassName)).toBe(true);
  });

  it("should not apply any class name when there is no label specified", () => {
    const mountWrapper = mount(
      <HvProvider>
        <HvCheckBox labelPlacement={labelPositions.start} />
      </HvProvider>
    ).find(HvCheckBox);
    expect(mountWrapper.html().includes(labelStartClassName)).toBe(false);
    expect(mountWrapper.html().includes(labelEndClassName)).toBe(false);
  });
});
