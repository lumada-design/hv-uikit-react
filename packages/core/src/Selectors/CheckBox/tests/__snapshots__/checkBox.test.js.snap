// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should render correctly 1`] = `
<WithStyles(HvCheckbox)
  id="test"
  inputProps={
    Object {
      "aria-label": "Checkbox example",
    }
  }
>
  <HvCheckbox
    classes={
      Object {
        "checkBox": "HvCheckBox-checkBox",
        "container": "HvCheckBox-container",
        "disableFocus": "HvCheckBox-disableFocus",
        "icon": "HvCheckBox-icon",
        "labelDisabled": "HvCheckBox-labelDisabled",
        "labelEnd": "HvCheckBox-labelEnd",
        "labelStart": "HvCheckBox-labelStart",
        "labelTypography": "HvCheckBox-labelTypography",
      }
    }
    id="test"
    inputProps={
      Object {
        "aria-label": "Checkbox example",
      }
    }
  >
    <WithStyles(ForwardRef(FormControlLabel))
      className="HvCheckBox-container"
      classes={
        Object {
          "disabled": "HvCheckBox-labelDisabled",
          "label": "HvCheckBox-labelTypography",
        }
      }
      control={
        <WithStyles(ForwardRef(Checkbox))
          checkedIcon={
            <WithStyles(CheckboxCheck)
              className="HvCheckBox-icon"
            />
          }
          className="HvCheckBox-checkBox"
          color="default"
          disableRipple={true}
          disabled={false}
          icon={
            <WithStyles(Checkbox)
              className="HvCheckBox-icon"
            />
          }
          id="test-input"
          indeterminateIcon={
            <WithStyles(CheckboxPartial)
              className="HvCheckBox-icon"
            />
          }
          inputProps={
            Object {
              "aria-label": "Checkbox example",
            }
          }
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
      }
      disabled={false}
      id="test"
      labelPlacement="end"
    >
      <ForwardRef(FormControlLabel)
        className="HvCheckBox-container"
        classes={
          Object {
            "disabled": "Mui-disabled HvCheckBox-labelDisabled",
            "label": "MuiFormControlLabel-label HvCheckBox-labelTypography",
            "labelPlacementBottom": "MuiFormControlLabel-labelPlacementBottom",
            "labelPlacementStart": "MuiFormControlLabel-labelPlacementStart",
            "labelPlacementTop": "MuiFormControlLabel-labelPlacementTop",
            "root": "MuiFormControlLabel-root",
          }
        }
        control={
          <WithStyles(ForwardRef(Checkbox))
            checkedIcon={
              <WithStyles(CheckboxCheck)
                className="HvCheckBox-icon"
              />
            }
            className="HvCheckBox-checkBox"
            color="default"
            disableRipple={true}
            disabled={false}
            icon={
              <WithStyles(Checkbox)
                className="HvCheckBox-icon"
              />
            }
            id="test-input"
            indeterminateIcon={
              <WithStyles(CheckboxPartial)
                className="HvCheckBox-icon"
              />
            }
            inputProps={
              Object {
                "aria-label": "Checkbox example",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          />
        }
        disabled={false}
        id="test"
        labelPlacement="end"
      >
        <label
          className="MuiFormControlLabel-root HvCheckBox-container"
          id="test"
        >
          <WithStyles(ForwardRef(Checkbox))
            checkedIcon={
              <WithStyles(CheckboxCheck)
                className="HvCheckBox-icon"
              />
            }
            className="HvCheckBox-checkBox"
            color="default"
            disableRipple={true}
            disabled={false}
            icon={
              <WithStyles(Checkbox)
                className="HvCheckBox-icon"
              />
            }
            id="test-input"
            indeterminateIcon={
              <WithStyles(CheckboxPartial)
                className="HvCheckBox-icon"
              />
            }
            inputProps={
              Object {
                "aria-label": "Checkbox example",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <ForwardRef(Checkbox)
              checkedIcon={
                <WithStyles(CheckboxCheck)
                  className="HvCheckBox-icon"
                />
              }
              className="HvCheckBox-checkBox"
              classes={
                Object {
                  "checked": "Mui-checked",
                  "colorPrimary": "MuiCheckbox-colorPrimary",
                  "colorSecondary": "MuiCheckbox-colorSecondary",
                  "disabled": "Mui-disabled",
                  "indeterminate": "MuiCheckbox-indeterminate",
                  "root": "MuiCheckbox-root",
                }
              }
              color="default"
              disableRipple={true}
              disabled={false}
              icon={
                <WithStyles(Checkbox)
                  className="HvCheckBox-icon"
                />
              }
              id="test-input"
              indeterminateIcon={
                <WithStyles(CheckboxPartial)
                  className="HvCheckBox-icon"
                />
              }
              inputProps={
                Object {
                  "aria-label": "Checkbox example",
                }
              }
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            >
              <WithStyles(ForwardRef(SwitchBase))
                checkedIcon={
                  <WithStyles(CheckboxCheck)
                    className="HvCheckBox-icon"
                    fontSize="default"
                  />
                }
                className="HvCheckBox-checkBox"
                classes={
                  Object {
                    "checked": "Mui-checked",
                    "disabled": "Mui-disabled",
                    "root": "MuiCheckbox-root",
                  }
                }
                color="default"
                disableRipple={true}
                disabled={false}
                icon={
                  <WithStyles(Checkbox)
                    className="HvCheckBox-icon"
                    fontSize="default"
                  />
                }
                id="test-input"
                inputProps={
                  Object {
                    "aria-label": "Checkbox example",
                    "data-indeterminate": false,
                  }
                }
                onBlur={[Function]}
                onChange={[Function]}
                type="checkbox"
                value=""
              >
                <ForwardRef(SwitchBase)
                  checkedIcon={
                    <WithStyles(CheckboxCheck)
                      className="HvCheckBox-icon"
                      fontSize="default"
                    />
                  }
                  className="HvCheckBox-checkBox"
                  classes={
                    Object {
                      "checked": "PrivateSwitchBase-checked-2 Mui-checked",
                      "disabled": "PrivateSwitchBase-disabled-3 Mui-disabled",
                      "input": "PrivateSwitchBase-input-4",
                      "root": "PrivateSwitchBase-root-1 MuiCheckbox-root",
                    }
                  }
                  color="default"
                  disableRipple={true}
                  disabled={false}
                  icon={
                    <WithStyles(Checkbox)
                      className="HvCheckBox-icon"
                      fontSize="default"
                    />
                  }
                  id="test-input"
                  inputProps={
                    Object {
                      "aria-label": "Checkbox example",
                      "data-indeterminate": false,
                    }
                  }
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                >
                  <WithStyles(ForwardRef(IconButton))
                    className="PrivateSwitchBase-root-1 MuiCheckbox-root HvCheckBox-checkBox"
                    color="default"
                    component="span"
                    disableRipple={true}
                    disabled={false}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    tabIndex={null}
                  >
                    <ForwardRef(IconButton)
                      className="PrivateSwitchBase-root-1 MuiCheckbox-root HvCheckBox-checkBox"
                      classes={
                        Object {
                          "colorInherit": "MuiIconButton-colorInherit",
                          "colorPrimary": "MuiIconButton-colorPrimary",
                          "colorSecondary": "MuiIconButton-colorSecondary",
                          "disabled": "Mui-disabled",
                          "edgeEnd": "MuiIconButton-edgeEnd",
                          "edgeStart": "MuiIconButton-edgeStart",
                          "label": "MuiIconButton-label",
                          "root": "MuiIconButton-root",
                          "sizeSmall": "MuiIconButton-sizeSmall",
                        }
                      }
                      color="default"
                      component="span"
                      disableRipple={true}
                      disabled={false}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      tabIndex={null}
                    >
                      <WithStyles(ForwardRef(ButtonBase))
                        centerRipple={true}
                        className="MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root HvCheckBox-checkBox"
                        component="span"
                        disableRipple={true}
                        disabled={false}
                        focusRipple={true}
                        onBlur={[Function]}
                        onFocus={[Function]}
                        tabIndex={null}
                      >
                        <ForwardRef(ButtonBase)
                          centerRipple={true}
                          className="MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root HvCheckBox-checkBox"
                          classes={
                            Object {
                              "disabled": "Mui-disabled",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiButtonBase-root",
                            }
                          }
                          component="span"
                          disableRipple={true}
                          disabled={false}
                          focusRipple={true}
                          onBlur={[Function]}
                          onFocus={[Function]}
                          tabIndex={null}
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root HvCheckBox-checkBox"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={null}
                          >
                            <span
                              className="MuiIconButton-label"
                            >
                              <input
                                aria-label="Checkbox example"
                                className="PrivateSwitchBase-input-4"
                                data-indeterminate={false}
                                disabled={false}
                                id="test-input"
                                onChange={[Function]}
                                type="checkbox"
                                value=""
                              />
                              <WithStyles(Checkbox)
                                className="HvCheckBox-icon"
                                fontSize="default"
                              >
                                <Checkbox
                                  className="HvCheckBox-icon"
                                  classes={
                                    Object {
                                      "root": "HvIconCheckbox-root",
                                      "rootL": "HvIconCheckbox-rootL",
                                      "rootM": "HvIconCheckbox-rootM",
                                      "rootS": "HvIconCheckbox-rootS",
                                      "rootXs": "HvIconCheckbox-rootXs",
                                    }
                                  }
                                  fontSize="default"
                                >
                                  <div
                                    className="HvCheckBox-icon HvIconCheckbox-root HvIconCheckbox-rootS"
                                    fontSize="default"
                                  >
                                    <svg
                                      focusable={false}
                                      height="16px"
                                      viewBox="0 0 16 16"
                                      width="16px"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        className="color0"
                                        d="M0 0h16v16H0z"
                                        fill="#FFFFFF"
                                      />
                                      <path
                                        className="color1"
                                        d="M0-.00291v16h16v-16zm15 15H1v-14h14z"
                                        fill="#414141"
                                      />
                                    </svg>
                                  </div>
                                </Checkbox>
                              </WithStyles(Checkbox)>
                            </span>
                          </span>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(IconButton)>
                  </WithStyles(ForwardRef(IconButton))>
                </ForwardRef(SwitchBase)>
              </WithStyles(ForwardRef(SwitchBase))>
            </ForwardRef(Checkbox)>
          </WithStyles(ForwardRef(Checkbox))>
          <WithStyles(ForwardRef(Typography))
            className="MuiFormControlLabel-label HvCheckBox-labelTypography"
            component="span"
          >
            <ForwardRef(Typography)
              className="MuiFormControlLabel-label HvCheckBox-labelTypography"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              component="span"
            >
              <span
                className="MuiTypography-root MuiFormControlLabel-label HvCheckBox-labelTypography MuiTypography-body1"
              />
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </label>
      </ForwardRef(FormControlLabel)>
    </WithStyles(ForwardRef(FormControlLabel))>
  </HvCheckbox>
</WithStyles(HvCheckbox)>
`;
